tests/components/test_discovery.py

```diff
@@ -128,3 +129,20 @@ def discover(netdisco):
     mock_discover.assert_called_with(
         hass, SERVICE_NO_PLATFORM, SERVICE_INFO,
         SERVICE_NO_PLATFORM_COMPONENT, BASE_CONFIG)
+
+
+@asyncio.coroutine
+def test_load_component_hassio(hass):
+    """Test load hassio component."""
+    os.environ['HASSIO'] = "FAKE_HASSIO"
```

Comment:
Please use `patch.dict` so that this test doesn't have side effects: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict